risc0/zkp/core/src/sha.rs

```diff
@@ -127,3 +90,89 @@ impl Debug for Digest {
         Ok(())
     }
 }
+
+impl Clone for Digest {
+    fn clone(&self) -> Digest {
+        Digest(self.0)
+    }
+}
+
+// An implementation that provides SHA-256 hashing services.
+pub trait ShaImpl: Clone + Debug {
```

Comment:
I'm a little confused when adding an `Impl` suffix to a `Trait`. I think a more idiomatic rust spelling would just be `pub trait Sha`.

Reply:
Done